tests/python/direct/test_python_direct.py

```diff
@@ -556,3 +562,59 @@ def fusion_func(fd: FusionDefinition) -> None:
             "FusionDefinition did not run correctly with profile enabled! Error: "
             + str(e)
         )
+
+
+def test_lru_cache():
+    inputs = [
+        torch.randn(2, 4, device="cuda"),
+        torch.randn(2, 4, device="cuda"),
+    ]
+
+    def create_fd1(fd: FusionDefinition):
+        t0 = fd.from_pytorch(inputs[0])
+        t1 = fd.from_pytorch(inputs[1])
+        t2 = fd.ops.div(t0, t1)
+        fd.add_output(t2)
+
+    def create_fd2(fd: FusionDefinition):
+        t0 = fd.from_pytorch(inputs[0])
+        t1 = fd.from_pytorch(inputs[1])
+        t2 = fd.ops.div(t1, t0)
+        fd.add_output(t2)
+
+    @LruFusionCache(max_fusions=10)
+    def create_fusion(select_first_fd: bool):
+        with FusionDefinition() as fd:
+            if select_first_fd:
+                create_fd1(fd)
+            else:
+                create_fd2(fd)
+        return fd
```

Comment:
A style nitpick, this example looks pretty strange to me.

Does the example look better with
```
    def create_fusion(create_fd):
        with FusionDefinition() as fd:
            create_fd(fd)
        return fd
```

So later we can call `create_fusion(create_fd1)`.

Reply:
Done